### PR TITLE
Fix error in agent when `use_memory` is `False`

### PR DESCRIPTION
**Pull request by DavdGao**
## Description

- Fix the error when `self.memory` is `None` in `reply` function. 
- Disable the default value of `model_config_name` of built-in agents, so that a model must be provided when using agent.

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: src/agentscope/agents/dialog_agent.py [hunk 1]
```diff
@@ -68,13 +68,13 @@ def reply(self, x: dict = None) -> dict:
             response to the user's input.
         """
         # record the input if needed
-        if x is not None:
+        if not self.memory:
```
Comment (qbc2016) [inline src/agentscope/agents/dialog_agent.py:71]: Should it be `if self.memory`? BTW, if use_memory is False, does `self.memory` defined?

Comment (DavdGao) [reply src/agentscope/agents/dialog_agent.py]: Thanks! The error is fixed

**Review by ZiTao-Li — APPROVED** [review]
lgtm

File: src/agentscope/agents/dialog_agent.py [hunk 2]
```diff
@@ -68,13 +68,13 @@ def reply(self, x: dict = None) -> dict:
             response to the user's input.
         """
         # record the input if needed
-        if x is not None:
+        if self.memory:
```
Comment (qbc2016) [inline src/agentscope/agents/dialog_agent.py:71]: If set `use_memory=False` when initializing an agent, `self.memory` is not defined, which will cause `AttributeError: 'DialogAgent' object has no attribute 'memory'`

Comment (DavdGao) [reply src/agentscope/agents/dialog_agent.py:71]: The bug is fixed accordingly.

**Pull request merged**